swri_roscpp/include/swri_roscpp/parameters.h

```diff
@@ -1,15 +1,30 @@
 #ifndef SWRI_ROSCPP_PARAMETERS_H_
 #define SWRI_ROSCPP_PARAMETERS_H_
 
+#include <algorithm>
+#include <set>
+#include <string>
+#include <vector>
+
+#include <boost/bind.hpp>
+
+#include <ros/console.h>
 #include <ros/node_handle.h>
 
 namespace swri
 {
+
+  /// This set stores all of the parameters that have been got with this library
+  /// It is used for the getUnusedParamKeys and WarnUnusedParams functions
+  std::set<std::string> _used_params;
```

Comment:
This either needs to be declared static or extern.  extern makes more sense to work well with nodes that are spread across several files, but means swri_roscpp will have to produce an actual library that is linked into dependent code.  I'd also request that the prefix be changed from _ to g_ but that's a personal preference.